aten/src/ATen/native/native_functions.yaml

```diff
@@ -3518,6 +3518,15 @@
   variants: method
   named_guard: False
 
+- func: result_type(Tensor tensor, Tensor other) -> ScalarType
+  variants: function
+
+- func: result_type(Tensor tensor, Scalar other) -> ScalarType
```

Comment:
so...it's not actually enforced that we have overload names?  @dzhulgakov / @smessmer is your current story that we just don't need to worry about it?